src/systemcmds/hardfault_log/hardfault_log.c

```diff
@@ -910,6 +1121,35 @@ __EXPORT int hardfault_write(char *caller, int fd, int format, bool rearm)
 	return ret;
 }
 
+int hardfault_store_ulog_filename(const char *log_file)
```

Comment:
Since the the hard fault logging is API file character based, and this is std file ops it can live in the logger.